cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -172,6 +201,130 @@ func (s *Server) GetAvailablePackages(ctx context.Context, request *corev1.GetAv
 	}, nil
 }
 
+// GetPackageMeta streams the package metadata based on the request.
+func (s *Server) GetPackageMeta(ctx context.Context, request *corev1.GetPackageMetaRequest) (*corev1.GetPackageMetaResponse, error) {
+	log.Infof("+GetPackageMeta()")
+
+	url, err := s.pullChartTarball(ctx, request.Package)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("Found chart url: [%s]", *url)
+
+	// unzip and untar .tgz file
+	meta, err := fetchMetaFromChartTarball(request.Package.Name, *url)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev1.GetPackageMetaResponse{
+		Meta: &corev1.GetPackageMetaResponse_PackageMeta{
+			Readme: meta[readme],
+		},
+	}, nil
+}
+
+func (s *Server) pullChartTarball(ctx context.Context, packageRef *corev1.PackageReference) (*string, error) {
+	client, err := s.GetClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceIfc := client.Resource(chartsResource).Namespace("default")
+
+	// see if we the chart already exists
+	chartList, err := resourceIfc.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO it'd be better if we could filter on server-side
```

Comment:
You should be able to use the `metav1.ListOptions{}` above to specify filtering using the [`FieldSelector`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ListOptions). More info at https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/ . You may even be able to specify that the pull should be complete to be included (ie. that status conditions ready is true), not sure, but that'd be nice.

Again, at this point it's fine if you just want to update your comment to link to that and do it in a later PR.

Reply:
As I remember I did try a few things here. The problem is the set of supported fields in FieldSelector is very small. things like spec.chart are certainly not supported. see https://github.com/kubernetes/client-go/issues/713 and https://github.com/flant/shell-operator/blob/8fa3c3b8cfeb1ddb37b070b7a871561fdffe788b/HOOKS.md#fieldselector. Nevertheless, I made a TODO comment here  just as you suggested to see if I can improve later